resource/order.go

```diff
@@ -119,41 +110,20 @@ func (oe OrderEndpoint) GetOrder(orderId int) (model.Order, error) {
 		return model.Order{}, errors.New(errorMessage)
 	}
 
-	if response.IsError() {
-		rawHead := response.GetHead()
-		errorMessage, err := jsonparser.GetString(rawHead, "ErrorMessage")
-
-		if err != nil {
-			return model.Order{}, err
-		}
-
-		return model.Order{}, errors.New(errorMessage)
-	}
-
 	rawBody := response.GetBody()
-	rawOrders, _, _, err := jsonparser.Get(rawBody, "Orders")
-
-	if err != nil {
+	var orders model.Orders
+	if err := json.Unmarshal(rawBody, &orders); nil != err {
 		return model.Order{}, err
 	}
 
-	rawOrder, _, _, err := jsonparser.Get(rawOrders, "Order")
-
-	if err != nil {
-		return model.Order{}, err
-	}
-
-	order := model.Order{}
-	if len(rawOrder) == 0 {
-		return model.Order{}, nil
+	if len(orders.Orders) == 1 {
```

Comment:
just for the sake of consistency check if len is 0 then return model.Order{}, errors.New("cannot find order") otherwise return orders.Orders[0], nil